app/controllers/consolidations/AssociateUCRConfirmationController.scala

```diff
@@ -20,17 +20,22 @@ import controllers.actions.{AuthenticatedAction, JourneyRefiner}
 import javax.inject.Inject
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.MovementRepository
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import views.html.associate_ucr_confirmation
 
+import scala.concurrent.ExecutionContext
+
 class AssociateUCRConfirmationController @Inject()(
   authenticate: AuthenticatedAction,
   getJourney: JourneyRefiner,
   mcc: MessagesControllerComponents,
+  movementRepository: MovementRepository,
   associateUCRConfirmPage: associate_ucr_confirmation
-) extends FrontendController(mcc) with I18nSupport {
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
 
-  def displayPage(): Action[AnyContent] = (authenticate andThen getJourney) { implicit request =>
-    Ok(associateUCRConfirmPage())
+  def displayPage(): Action[AnyContent] = (authenticate andThen getJourney).async { implicit request =>
+    movementRepository.removeByPid(request.pid).map(_ => Ok(associateUCRConfirmPage()))
```

Comment:
I'd argue the repository code should be part of the SubmissionService

Reply:
So in that case you can access confirmation page because you have empty cache.

I think that we can just use confirmation page in Summary Controller without separte endpoint or controller for it...

Reply:
Are you saying that we don't need this controller?  That the AssociateUCRSummaryController should just display the confirmation page (after the service has removed the cache)?

That's what the movement SummaryController does so it would be good to be consistent.  (The Summary controller also calls 'remove' on the repo - so maybe that should move to the service after successful submit as well.

Reply:
Yeah, I added sticker to tech board to display confirmation directly in submit method in Summary page, without creating Confirmation Controller

Reply:
So are going to do that now for this service and port solution back to movements?  Or do both as tech debt?

I don't mind.